build/wd_rust_crate.bzl

```diff
@@ -7,22 +7,40 @@ def rust_cxx_bridge(
         deps = [],
         visibility = [],
         strip_include_prefix = None,
-        include_prefix = None):
+        include_prefix = None,
+        needs_jsg = False):
+    # Prepare outputs and sources
+    outs = [src + ".h", src + ".cc"]
+    srcs = [src]
+
+    # If JSG is needed, add jsg.h to outputs and copy it from the generic header
+    if needs_jsg:
+        outs.append("jsg.h")
```

Comment:
As mentioned via chat: Instead of copying headers around (feels scary), you should be able to just add `@workerd//src/workerd/jsg` to deps here.

Reply:
I would go even farther: remove needs_jsg and make targets specify jsg in `deps`